sysl2/sysl/ints_view.go

```diff
@@ -0,0 +1,587 @@
+					fmt.Fprintln(v.stringBuilder)
+					callsDrawn[*apps] = struct{}{}
+				}
+			}
+		}
+		for _, app := range params.apps {
+			for _, mixin := range v.mod.Apps[app].GetMixin2() {
+				mixinName := strings.Join(mixin.Name.Part, " :: ")
+				fmt.Fprintf(
+					v.stringBuilder,
+					"%s <|.. %s",
+					v.VarManagerForComponent(mixinName, nameMap),
+					v.VarManagerForComponent(app, nameMap),
+				)
+				fmt.Fprintln(v.stringBuilder)
+			}
+		}
+	}
+}
+
+func (v *IntsDiagramVisitor) drawSystemView(viewParams viewParams, params *IntsParam, nameMap map[string]string) {
+	callsDrawn := map[CallsDrawn]struct{}{}
+	for _, dep := range params.integrations {
+		appA := dep.Self.Name
+		appB := dep.Target.Name
+		apps := &CallsDrawn{
+			Self:   appA,
+			Target: appB,
+		}
+		var direct []string
+		if _, ok := params.drawableApps[appA]; ok {
+			direct = append(direct, appA)
+		}
+		if _, ok := params.drawableApps[appB]; ok {
+			direct = append(direct, appB)
+		}
+		appA = strings.Split(appA, " :: ")[0]
+		appB = strings.Split(appB, " :: ")[0]
+		_, ok := callsDrawn[*apps]
+		if appA != appB && !ok {
+			if direct != nil || viewParams.indirectArrowColor != ArrowColorNone {
+				indirect := ""
+				if direct == nil {
+					indirect = " <<indirect>>"
+				}
+				fmt.Fprintf(
+					v.stringBuilder,
+					"%s --> %s%s",
+					v.VarManagerForComponent(appA, nameMap),
+					v.VarManagerForComponent(appB, nameMap),
+					indirect,
+				)
+				fmt.Fprintln(v.stringBuilder)
+				callsDrawn[*apps] = struct{}{}
+			}
+		}
+	}
+}
+
+func (v *IntsDiagramVisitor) generateComponentView(args *Args, viewParams viewParams, params *IntsParam) string {
+
+	v.stringBuilder.WriteString(`@startuml
+`)
+	if viewParams.diagramTitle != "" {
+		fmt.Fprintln(v.stringBuilder, "title "+viewParams.diagramTitle)
+	}
+	v.stringBuilder.WriteString(`hide stereotype
+scale max 16384 height
+skinparam component {
+  BackgroundColor FloralWhite
+  BorderColor Black
+  ArrowColor Crimson
+`)
+	if viewParams.highLightColor != "" {
+		fmt.Fprintln(v.stringBuilder, "  BackgroundColor<<highlight>> "+viewParams.highLightColor)
+	}
+	if viewParams.arrowColor != "" {
+		fmt.Fprintln(v.stringBuilder, "  ArrowColor "+viewParams.arrowColor)
+	}
+
+	if viewParams.indirectArrowColor != "" && viewParams.indirectArrowColor != ArrowColorNone {
+		fmt.Fprintln(v.stringBuilder, "  ArrowColor<<indirect>> "+viewParams.indirectArrowColor)
+	}
+	v.stringBuilder.WriteString(`}
+`)
+	nameMap := map[string]string{}
+	if args.clustered || viewParams.endptAttrs["view"].GetS() == "clustered" {
+		nameMap = v.buildClusterForComponentView(params.apps)
+	}
+	v.drawComponentView(viewParams, params, nameMap)
+	v.stringBuilder.WriteString("@enduml")
+	return v.stringBuilder.String()
+}
+
+func GenerateView(args *Args, params *IntsParam, mod *sysl.Module) string {
+	var stringBuilder strings.Builder
+	var titleParser *FormatParser
+	v := MakeIntsDiagramVisitor(mod, &stringBuilder, params.drawableApps, args.project)
```

Comment:
do you really need to pass `	var stringBuilder strings.Builder` to `MakeIntsDiagramVisitor`?

the only reason you seem to need this in GenerateView is to add the header a the start.

Move ALL these constant strings to the top the file like so...

```
const PUML_HEADER =`''''''''''''''''''''''''''''''''''''''''''
''                                      ''
''  AUTOGENERATED CODE -- DO NOT EDIT!  ''
''                                      ''
''''''''''''''''''''''''''''''''''''''''''
 `

Reply:
moved all constant string